org-code-javabuilder/lib/src/main/java/org/code/javabuilder/LambdaRequestHandler.java

```diff
@@ -173,6 +186,8 @@ public String handleRequest(Map<String, String> lambdaInput, Context context) {
     } finally {
       // Stop timeout listener and clean up
       timeoutNotifierThread.interrupt();
+      PerformanceTracker.getInstance().trackInstanceEnd();
+      PerformanceTracker.getInstance().logPerformance();
```

Comment:
Do we need to call these methods anytime before we call `System.exit()` as well? I think we'd just need add these in `AWSSystemExitHelper` and here above line 160. However, I can also see us intentionally not report these if we want to take their absence to mean an abnormal shutdown?

Reply:
Oh yes, we definitely want this to be considered part of the cleanup logic. I'll go ahead and add this to those two areas. This also is a pretty clear indicator that our error handlining and cleanup logic has fractured pretty significantly. I went ahead and moved the [cleanup task](https://codedotorg.atlassian.net/browse/JAVA-531) up to "Up Next."